Hoping to Merge Develop Changes to Stable

drnimbusrain opened the pull request:
@zmoon

drnimbusrain committed:
Updated python script due to txt output file name changes.

drnimbusrain committed:
Added new time stamp suffix to txt output for python script.

zmoon committed:
Timed text files

but not parsing the time from the file name yet

zmoon committed:
Timed nc files

drnimbusrain committed:
Revised time step/stamp outputs for TXT and NC.

drnimbusrain committed:
Updated README

drnimbusrain committed:
Updated README

drnimbusrain committed:
Fixed write_txt error for missing argument.

drnimbusrain committed:
Changed python/canopy_app.py for new text headers.

drnimbusrain committed:
Trying this way.

drnimbusrain committed:
Updated skiprows.

drnimbusrain committed:
Changed time units to "seconds since"

zmoon committed:
Read time stamp from text files

zmoon committed:
Fix nc reading for now

zmoon committed:
Merge branch 'feature/timestep' of https://github.com/noaa-oar-arl/canopy-app into feature/timestep

zmoon committed:
Back to decoding nc times

drnimbusrain committed:
Fixed output time.

drnimbusrain committed:
Merge branch 'feature/timestep' of https://github.com/noaa-oar-arl/canopy-app into feature/timestep

quaz115 committed:
Leaf Age Response function for all Biogenic VOCs consistent with MEGAN

quaz115 committed:
Compiles now with LEAFAGE specific changes, findent also resolved

quaz115 committed:
Update namelist.canopy (leafage_opt set to 0)

quaz115 committed:
LEAFAGE uses TEMP2 for TABOVECANOPY (fixed)

quaz115 committed:
Merge remote-tracking branch 'origin/develop' into develop

quaz115 committed:
temp2 replaced with tmp2mref in canopy_calcs.F90 and Tt with TABOVE in canopy_utils_mod.F90 in the GET_GAMMA_LEAFAGE function

quaz115 committed:
Naming conventions imprved and README description of leafage_opt added

quaz115 committed:
README leafage_opt description updated

angehung5 committed:
Update global_data_process.py

drnimbusrain committed:
Merge pull request #93 from angehung5/main

Update global_data_process.py

drnimbusrain committed:
Changed NL back to 3 times and added point example.

zmoon committed:
Keep default case in CI as single file

zmoon committed:
Fix setup for multiple input files

zmoon committed:
Deal with multiple times in output txt files

zmoon committed:
ntime

drnimbusrain committed:
Updated point files to change vtype and fix fch.

drnimbusrain committed:
Merge branch 'feature/timestep' of https://github.com/noaa-oar-arl/canopy-app into feature/timestep

zmoon committed:
Replace fill value with NaN when reading txt outputs

zmoon committed:
Fix examples

- need to specify ntime now
- different canopy height for point
- for cases generation, limit to single time step for now
  - since file_vars can be str OR list(str) now, hard to know
    if it is list of options or a single config

only one time since CI currently overriding default nl

recall squeezed before returning

drnimbusrain committed:
Updated SE txt and nc files for LAI updates from Wei-Ting.

drnimbusrain committed:
Removed timestamp from updated default SE txt files.

drnimbusrain committed:
Merge pull request #87 from noaa-oar-arl/feature/timestep

Adding new Feature/timestep

drnimbusrain committed:
Merge branch 'develop' into feature/frp_flameh

drnimbusrain committed:
Merge branch 'develop' into feature/pavd

drnimbusrain committed:
Replaced missing userdef, flameh_frp, from conflict.

quaz115 committed:
Merge branch 'develop' into develop

quaz115 committed:
Resolved LAIpast=LAIcurrent Warning

quaz115 committed:
TABOVECANOPY removed as input parameter from canopy_bio

quaz115 committed:
LEAFAGE merged with TIMEFEATURE (LAI across tsteps print diagnostics tested as well)

zmoon committed:
Add point example with leaf age opt to nb

angehung5 committed:
Update README.md

angehung5 committed:
Update namelist.canopy

angehung5 committed:
Update canopy_calcs.F90

angehung5 committed:
Update canopy_canopts_mod.F90

angehung5 committed:
Update canopy_fire_mod.F90

angehung5 committed:
Update canopy_readnml.F90

angehung5 committed:
Update canopy_utils_mod.F90

drnimbusrain committed:
Merge pull request #88 from noaa-oar-arl/feature/frp_flameh

Added vegtype dependent FRP to FlameH calculations.

quaz115 committed:
Interpolation of Monthly/daily (user defined) LAI input to Model  (hourly) time step

zmoon committed:
Add Zenodo DOI badge that resolves to latest

quaz115 committed:
lai_tstep as variable name, A* factors removed from canopy_bio call as they are local vars, leafage_opt=1 (off, gamma_leafage=1) now default, other README and namelist.canopy edits

quaz115 committed:
Warning changed to error for non-daily/monthly LAI input for now

drnimbusrain committed:
Merge branch 'develop' into feature/pavd

drnimbusrain committed:
Fixed warning on unused dummy argument.

drnimbusrain committed:
Fixed canopy_calcs too.

angehung5 committed:
Update daily LAI

angehung5 committed:
Update txt file

drnimbusrain committed:
Update README.md

Updating README for daily example files from AWS under Table 2.

angehung5 committed:
Update global_data_process.py

angehung5 committed:
Update Table 2

Add pavd and data time frequency

angehung5 committed:
Update Table 2

Add pavd and data time frequency

angehung5 committed:
Update global_data_process.py

angehung5 committed:
Update global_data_process.py

angehung5 committed:
Update global_data_process.py

angehung5 committed:
Update global_data_process.py

angehung5 committed:
Update global_data_process.py

drnimbusrain committed:
Update README.md

drnimbusrain committed:
Merge pull request #95 from angehung5/main

Add pavd and daily LAI

angehung5 committed:
Add lev to SE file

angehung5 committed:
Delete input/canopy.t12z.20220630.testf023.southeast.nc

angehung5 committed:
Delete input/canopy.t12z.20220701.testf000.southeast.nc

angehung5 committed:
Delete input/canopy.t12z.20220701.testf001.southeast.nc

angehung5 committed:
Add lev

angehung5 committed:
Delete input/gfs.t12z.20220701.sfcf001.scanopy.nc

angehung5 committed:
Add lev

angehung5 committed:
update PAVD

angehung5 committed:
update PAVD

quaz115 committed:
No interpolation and daily LAI inputs used: SE US test done

drnimbusrain committed:
Updated python README and slurm script for global runs.

drnimbusrain committed:
Updated python/README and slurm script for global.

drnimbusrain committed:
Updated for global python and run examples.

drnimbusrain committed:
Updated for global example scripts.

drnimbusrain committed:
Updated python/README for global run example.

quaz115 committed:
Update input/namelist.canopy leaving leafage_opt =1 (OFF) as default in namelist.canopy

Co-authored-by: Patrick Campbell <dr.nimbusrain@gmail.com>

quaz115 committed:
Update src/canopy_calcs.F90  Remove print diagnostics

Co-authored-by: Patrick Campbell <dr.nimbusrain@gmail.com>

quaz115 committed:
Update src/canopy_canopts_mod.F90 corrected description in canopy_canopts_mod.F90

Co-authored-by: Patrick Campbell <dr.nimbusrain@gmail.com>

quaz115 committed:
leaving leafage_opt =1 (OFF) as default

quaz115 committed:
leaving leafage_opt =1 (OFF) as default and tstep docstring description corrected in canopy_canopts_mod and REmove print diags

quaz115 committed:
Pre-commit passed

quaz115 committed:
Try again with latest updates

quaz115 committed:
namelist.canopy reverted to SE US options

quaz115 committed:
namelist.canopy clean up final

quaz115 committed:
namelist.canopy input files rveerted to match develop

quaz115 committed:
Cleaned namelist.canopy input files to match develop

quaz115 committed:
.txt commented input list added: Cleaned namelist.canopy input files to match develop

drnimbusrain committed:
Merge pull request #92 from quaz115/develop

Leaf Age Response function for all Biogenic VOCs consistent with MEGAN

drnimbusrain committed:
Initial commits of restructured NL and README for clarity.

drnimbusrain committed:
Updated README.

drnimbusrain committed:
Update README

drnimbusrain committed:
Update README

drnimbusrain committed:
Updated README

drnimbusrain committed:
Updated README

drnimbusrain committed:
Also added python/README and example slurm script for global.

drnimbusrain committed:
Removed global example and slurm scripts, better in separate PR.

angehung5 committed:
Merge branch 'noaa-oar-arl:develop' into main

angehung5 committed:
daily LAI and PAVD

angehung5 committed:
Update README.md

angehung5 committed:
Update README.md

angehung5 committed:
Allow multiple time inputs

angehung5 committed:
Update global_data_process.py

angehung5 committed:
Update global_data_process.py

angehung5 committed:
Update README.md

angehung5 committed:
Update global_data_process.py

angehung5 committed:
Update README.md

angehung5 committed:
Update global_data_process.py

angehung5 committed:
Update global_data_process.py

angehung5 committed:
Update global_data_process.py

angehung5 committed:
Update global_data_process.py

drnimbusrain committed:
Merge pull request #97 from angehung5/main

Update example files and global process script

drnimbusrain committed:
Updated example slurm and python global for new updates.

angehung5 committed:
Update README.md

change required cpu time and mem for global script

angehung5 committed:
Update run_canopy_slurm.sh

change required cpu time and mem

angehung5 committed:
Update README.md

angehung5 committed:
Update run_canopy_slurm.sh

drnimbusrain committed:
Updated python/README to be specific to Example 1 global processing

drnimbusrain committed:
Merge branch 'feature/nl_format' of https://github.com/noaa-oar-arl/canopy-app into feature/nl_format

drnimbusrain committed:
Merge branch 'develop' into feature/nl_format

drnimbusrain committed:
Updated python README

angehung5 committed:
Update README.md

angehung5 committed:
Update output filename

angehung5 committed:
Update run_canopy_slurm.sh

angehung5 committed:
Update README.md

angehung5 committed:
Update run_canopy_slurm.sh

angehung5 committed:
Update README.md

drnimbusrain committed:
Update python/README.md

drnimbusrain committed:
Merge pull request #98 from noaa-oar-arl/slurm_test

Update slurm requirements

drnimbusrain committed:
Merge branch 'develop' into feature/pavd

drnimbusrain committed:
Testing new PAVD and adding some conditions.

angehung5 committed:
Add files via upload

drnimbusrain committed:
Removed .nc SE test files.

drnimbusrain committed:
Adding updated SE .nc files back.

drnimbusrain committed:
Turned debug prints on for i=25 and j=25.

drnimbusrain committed:
Commented out debug prints and updated README.

drnimbusrain committed:
Updated NL for PAVD options.

drnimbusrain committed:
Adding clarifications and TODOs for upcoming PAVD text inputs.

drnimbusrain committed:
Hotfix for bug with no leafage current/previous/timestep for LAI when using .txt inputs.

drnimbusrain committed:
Merge branch 'develop' into feature/pavd

angehung5 committed:
Update CH

angehung5 committed:
Delete input/gfs.t12z.20220630.sfcf023.canopy.txt

angehung5 committed:
Delete input/gfs.t12z.20220701.sfcf000.canopy.txt

angehung5 committed:
Delete input/gfs.t12z.20220701.sfcf001.canopy.txt

angehung5 committed:
Reupload text file

angehung5 committed:
Update NAU CH

angehung5 committed:
Update point_file_20220630.sfcf023.txt

angehung5 committed:
Update point_file_20220701.sfcf000.txt

angehung5 committed:
Update point_file_20220701.sfcf001.txt

angehung5 committed:
Update point file

angehung5 committed:
Update NAU CH

angehung5 committed:
Delete input/canopy.t12z.20220630.testf023.southeast.txt

angehung5 committed:
Delete input/canopy.t12z.20220701.testf000.southeast.txt

angehung5 committed:
Delete input/canopy.t12z.20220701.testf001.southeast.txt

angehung5 committed:
Change vtype to fit fch

angehung5 committed:
Change vtype to fit fch

angehung5 committed:
Change vtype to fit fch

angehung5 committed:
add point pavd text file

angehung5 committed:
add point pavd text file

drnimbusrain committed:
New functionality to read/test PAVD from text file.

drnimbusrain committed:
Changing NL back to defaults.

angehung5 committed:
Update CH

angehung5 committed:
Update FH data source

angehung5 committed:
Update FH

angehung5 committed:
Update FH

angehung5 committed:
Update FH

angehung5 committed:
Update FH and vtype

angehung5 committed:
Update a new point

angehung5 committed:
Update a new point

angehung5 committed:
Update a new point

angehung5 committed:
Update a new point

angehung5 committed:
Update a new point

angehung5 committed:
Update a new point

drnimbusrain committed:
Initial changes to PAVD for shape distribution conversion.

drnimbusrain committed:
Merge branch 'feature/pavd' of https://github.com/noaa-oar-arl/canopy-app into feature/pavd

drnimbusrain committed:
Updating PAVD method to use only zcanmax.

drnimbusrain committed:
Quick fix to remove Massman aspen type from evergreen.

drnimbusrain committed:
Merge branch 'develop' into feature/pavd

drnimbusrain committed:
New WAI adaptation for PAI calculation option from Fang et al.

drnimbusrain committed:
Adding CLU to PAI calculation and better description.

drnimbusrain committed:
Merge pull request #100 from noaa-oar-arl/feature/wai2pai

New WAI adaptation for PAI calculation option from Fang et al.

drnimbusrain committed:
Merge branch 'develop' into feature/pavd

angehung5 committed:
Edit the description of pavd_set

drnimbusrain committed:
Added PAVD condition if ZK (at max GEDI PAVD height) > GEDI FCH.

drnimbusrain committed:
Merge pull request #91 from noaa-oar-arl/feature/pavd

Adding new Feature/pavd

drnimbusrain committed:
Fixed derivation of PAI formulation, now based only on alpha.

drnimbusrain committed:
Merge branch 'develop' of https://github.com/noaa-oar-arl/canopy-app into develop

drnimbusrain committed:
Updated alpha values based on internal paper data from Figure 1 of Fang et al. (2019).

drnimbusrain committed:
Separated SSG from crops in foliage distribution and added croplands/mosaic and tundra cats.

drnimbusrain committed:
Merge pull request #101 from noaa-oar-arl/ssg_opts

Separated SSG from crops in foliage distribution and other vtypes.

drnimbusrain committed:
Added check for directory exists in python/global_data_process.

drnimbusrain committed:
Needed limitation in canopy sublayers for tall canopies.

angehung5 committed:
Change YY to 2022 for gfs.canopy

drnimbusrain committed:
Merge pull request #103 from noaa-oar-arl/fix_global_script

Change YY to 2022 for gfs.canopy in global script

drnimbusrain committed:
Fixed issue of tall canopies leading to layer issue.

drnimbusrain committed:
Added more gfortran debug flag option and fixed FPEs.

zmoon committed:
Deal with xarray from pandas time warning

by converting to ns precision beforehand,
as the warning suggests

drnimbusrain committed:
Removed Aspen from distributions, clean print, and fixed FPE for point data.

zmoon committed:
Fix var select canopy mean

the issue seems to be that canopy height has a time dim
in the default case now, so `da.z <= hc` was re-introducing
a time dimension

zmoon committed:
Add units to some plots in 1-D section

zmoon committed:
Guard hc time sel

angehung5 commented on README.md:
```diff
@@ -149,10 +150,12 @@ The Canopy-App input data in [Table 2](#table-2-canopy-app-required-input-variab
 | `prate_ave`                      | Average mass precipitation rate (kg m-2 s-1) | UFS NOAA/GFSv16                                   |
 | **External Canopy Variables**    | **Variable Description and Units**          | **Data Source/Reference (if necessary)**           |
-| `fh`                             | Forest canopy height (m)                    | Fused GEDI/Landsat data. Data Period=2020. ([Potapov et al., 2020](https://doi.org/10.1016/j.rse.2020.112165)) |
-| `clu`                            | Forest clumping index (dimensionless)       | GriddingMachine/MODIS. Data Period=2001-2017 Climatology. ([Wei et al., 2019](https://doi.org/10.1016/j.rse.2019.111296)). Extended globally for high latitudes using methods described [here](https://gmuedu-my.sharepoint.com/:w:/g/personal/whung_gmu_edu/EdglXmW2kzBDtDj1xV0alGcB1Yo2I8hzdyWGVGB2YOTfgw). |
```
Maybe use "Vegetation clumping index"? It should be clu for all species.

drnimbusrain replied:
Agreed and changed to canopy clumping index.

angehung5 reviewed (approved):
I did not check the bioemi and ipynb scripts. Others look good. But I suggest changing "Forest clumping index" to "Vegetation clumping index" in README, since it should include all vegetation species.

drnimbusrain commented:
Thank you Wei-Ting! This brings up the question again on if we should make
the change from MODIS forest fraction to total vegetation fraction.  

On Wed, Jan 24, 2024, 4:55 PM Wei-Ting Hung ***@***.***>
wrote:

> ***@***.**** approved this pull request.
>
> I did not check the bioemi and ipynb scripts. Others look good. But I
> suggest changing "Forest clumping index" to "Vegetation clumping index" in
> README, since it should include all vegetation species.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/noaa-oar-arl/canopy-app/pull/104#pullrequestreview-1842455913>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGLFYNUG66SFQFXQ5UXBMWLYQF7N7AVCNFSM6AAAAABCIA7OC2VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTQNBSGQ2TKOJRGM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

drnimbusrain commented:
@angehung5  Did some checking, currently the ffrac is only used to define [contiguous canopy grid cell threshold](https://github.com/noaa-oar-arl/canopy-app/blob/develop/src/canopy_calcs.F90#L139), and in the [PAI calculation option as a surrogate for canopy cover fraction](https://github.com/noaa-oar-arl/canopy-app/blob/develop/src/canopy_utils_mod.F90#L71) (but this M17 formulation/option leads to results we do not trust).   

Overall, since canopy-app is technically being driven by any canopy type (for height, lai, clu, etc.) it seems to me that we should change this "_ffrac_" variable into our canopy-app inputs to rather use a total canopy fraction, i.e.,  "_canfrac_".  It would be more general, and likely more representative as a threshold (and in the PAI M17 calculation).  We could add this canfrac also as an additional 1-km global file for UFS later too.

I don't think this effects much your paper results, but could be something that could be brought in here.  As you mentioned, this is something you already have as the annual total vegetation fraction (TVFRAC) in your [global extension document](https://gmuedu-my.sharepoint.com/:w:/g/personal/whung_gmu_edu/EdglXmW2kzBDtDj1xV0alGcB1Yo2I8hzdyWGVGB2YOTfgw), and something that was needed for ffrac anyway.

What do you think?

drnimbusrain committed:
Changed README forest to canopy for clumping index description.

drnimbusrain committed:
Merge pull request #105 from noaa-oar-arl/fix/nb

Small fixes to the examples notebook

drnimbusrain commented:
@angehung5  See issue #106   I think we should also make this change for 'ffrac' to 'canfrac' before merging develop to stable...

zmoon committed:
Merge branch 'develop' into update-ci

Initial DEBUG=2 option

some tweaks to flags, including setting error limit 0
for DEBUG=2

zmoon committed:
fmt

zmoon committed:
Gamma leaf age calc only if on

with the enhanced debug flags, was complaining about the
past (and/or current) LAI being unitialized in the first
expression of the calculation

zmoon committed:
DEBUG=2 in readme

zmoon committed:
Update actions/checkout

drnimbusrain committed:
Merge pull request #85 from noaa-oar-arl/update-ci

Update CI

drnimbusrain committed:
Fixed array temporary warnings by explicitly creating the arrays.

drnimbusrain committed:
Merge pull request #107 from noaa-oar-arl/array_temp

Fixed array temporary warnings by explicitly creating the arrays.

drnimbusrain committed:
Checked all code paths/options and fixed FPEs.

drnimbusrain committed:
Merge pull request #108 from noaa-oar-arl/fix_fpes

Checked all code paths/options and fixed FPEs.

angehung5 committed:
Update ch and canfrac

angehung5 committed:
Delete input/point_file_20220630.sfcf001.txt

angehung5 committed:
Update ch and canfrac

angehung5 committed:
Update ch and canfrac

angehung5 committed:
Update ch and canfrac

angehung5 committed:
Update fh to ch

angehung5 committed:
Update fh to ch

angehung5 committed:
Update ffrac to canfrac

angehung5 committed:
Update ffrac to canfrac

angehung5 committed:
Update fh to ch

angehung5 committed:
Update fch to ch, ffrac to canfrac

angehung5 committed:
Update fh to ch, ffrac to canfrac

angehung5 committed:
Update ch_thresh and cf_thresh

angehung5 committed:
Update ch_thresh and cf_thresh

angehung5 committed:
Update ch_thresh and cf_thresh

angehung5 committed:
Update ch_thresh and cf_thresh

angehung5 committed:
Add newline to end of file

angehung5 committed:
Add newline to end of file

angehung5 committed:
Add newline to end of file

angehung5 committed:
Update ch_thresh and cf_thresh

angehung5 committed:
Update ch, canfrac, cf_thresh, ch_thresh

drnimbusrain committed:
Merge pull request #109 from angehung5/main

Update ch and canfrac
